Net07.HW/Net07.DynamicProgrammingAndClasses/Net07.DynamicProgrammingAndClasses/Square.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace Net07.DynamicProgrammingAndClasses
+{
+    class Square : AngleShapes
+    {
+        Point[] points;
+        public Square(Point a, Point b, Point c, Point d) : base(a, b, c, d) { }
+        public Square(Point[] array) : base(array)
+        {
+            if (array.Length != 4)
+            {
+                throw new ArgumentException($"Недопустимое число точек задаваемых для {this}");
```

Comment:
Because we have no override of `ToString` in this class, the message will be something like:
```
Недопустимое число точек задаваемых для Net07.DynamicProgrammingAndClasses.Square 
```

We should make `override ToString()` to customize this string